### ISSUE TITLE
Can not run gradle integration tests from a plugin directory

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#17043]
With #16989, we can not run anymore `gradle check` from a plugin directory:

```sh
cd plugins/ingest-attachment
gradle check
```

gives:

```
...
:plugins:ingest-attachment:integTest#copyPlugins
:plugins:ingest-attachment:integTest#installIngestAttachmentPlugin
Plugins directory [/Users/dpilato/Documents/Elasticsearch/dev/es-gradle/elasticsearch/plugins/ingest-attachment/build/cluster/integTest node0/elasticsearch-5.0.0-SNAPSHOT/plugins] does not exist. Creating...
-> Downloading file:/Users/dpilato/Documents/Elasticsearch/dev/es-gradle/elasticsearch/plugins/ingest-attachment/build/cluster/integTest node0/plugins tmp/ingest-attachment-5.0.0-SNAPSHOT.zip
Exception in thread "main" java.lang.IllegalArgumentException: illegal version format - snapshots are only supported until version 2.x
        at org.elasticsearch.Version.fromString(Version.java:152)
        at org.elasticsearch.plugins.PluginInfo.readFromProperties(PluginInfo.java:98)
        at org.elasticsearch.plugins.InstallPluginCommand.verify(InstallPluginCommand.java:255)
        at org.elasticsearch.plugins.InstallPluginCommand.install(InstallPluginCommand.java:313)
        at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:142)
        at org.elasticsearch.common.cli.CliTool.execute(CliTool.java:143)
        at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:74)
:plugins:ingest-attachment:integTest#installIngestAttachmentPlugin FAILED
```
 
The work-around for now is to run:

```sh
gradle :plugins:ingest-attachment:check
```

Is this something we can fix?

**Comment by s1monw** [comment]
works for me, did you try to turn it off and on again?

```BASH
panthor:elasticsearch simon$ cd plugins/ingest-attachment
panthor:ingest-attachment simon$ gradle check
:buildSrc:compileJava UP-TO-DATE
:buildSrc:compileGroovy UP-TO-DATE
:buildSrc:writeVersionProperties
:buildSrc:processResources
:buildSrc:classes
:buildSrc:jar
:buildSrc:sourcesJar UP-TO-DATE
:buildSrc:signArchives SKIPPED
:buildSrc:assemble
:buildSrc:compileTestJava UP-TO-DATE
:buildSrc:compileTestGroovy UP-TO-DATE
:buildSrc:processTestResources UP-TO-DATE
:buildSrc:testClasses UP-TO-DATE
:buildSrc:test UP-TO-DATE
:buildSrc:check UP-TO-DATE
:buildSrc:build
=======================================
Elasticsearch Build Hamster says Hello!
=======================================
  Gradle Version        : 2.10
  OS Info               : Mac OS X 10.11.1 (x86_64)
  JDK Version           : Oracle Corporation 1.8.0_66 [Java HotSpot(TM) 64-Bit Server VM 25.66-b17]
:core:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:core:processResources UP-TO-DATE
:core:classes
:core:jar
:plugins:ingest-attachment:compileJava
:plugins:ingest-attachment:processResources UP-TO-DATE
:plugins:ingest-attachment:classes
:plugins:ingest-attachment:checkstyleMain
:plugins:ingest-attachment:copyPluginPropertiesTemplate
:plugins:ingest-attachment:pluginProperties
:test:framework:compileJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
:test:framework:processResources UP-TO-DATE
:test:framework:classes
:test:framework:jar
:plugins:ingest-attachment:compileTestJava
:plugins:ingest-attachment:processTestResources
:plugins:ingest-attachment:testClasses
:plugins:ingest-attachment:checkstyleTest
:plugins:ingest-attachment:forbiddenApisMain
:plugins:ingest-attachment:forbiddenApisTest
:plugins:ingest-attachment:forbiddenApis
:plugins:ingest-attachment:checkstyle
:plugins:ingest-attachment:dependencyLicenses
:plugins:ingest-attachment:forbiddenPatterns
:plugins:ingest-attachment:jarHell
:plugins:ingest-attachment:licenseHeaders
:plugins:ingest-attachment:namingConventions
:plugins:ingest-attachment:thirdPartyAudit
:plugins:ingest-attachment:precommit
:plugins:ingest-attachment:test
==> Test Info: seed=446089BDC7AA2D47; jvms=3; suites=4
HEARTBEAT J2 PID(40949@panthor.local): 2016-03-10T10:58:05, stalled for 11.5s at: TikaDocTests.testFiles
Slow Tests Summary:
 22.91s | org.elasticsearch.ingest.attachment.TikaDocTests
 13.59s | org.elasticsearch.ingest.attachment.AttachmentProcessorTests
  4.64s | org.elasticsearch.ingest.attachment.AttachmentProcessorFactoryTests
  1.99s | org.elasticsearch.ingest.attachment.TikaImplTests

==> Test Summary: 4 suites, 18 tests
:plugins:ingest-attachment:jar
:plugins:ingest-attachment:bundlePlugin
:rest-api-spec:compileJava UP-TO-DATE
:rest-api-spec:processResources
:rest-api-spec:classes
:rest-api-spec:jar
:plugins:ingest-attachment:copyRestSpec
:distribution:integ-test-zip:buildZip
:plugins:ingest-attachment:integTest#clean
:plugins:ingest-attachment:integTest#checkPrevious SKIPPED
:plugins:ingest-attachment:integTest#stopPrevious SKIPPED
:plugins:ingest-attachment:integTest#extract
:plugins:ingest-attachment:integTest#configure
:plugins:ingest-attachment:integTest#copyPlugins
:plugins:ingest-attachment:integTest#installIngestAttachmentPlugin
:plugins:ingest-attachment:integTest#start
:plugins:ingest-attachment:integTest#wait
:plugins:ingest-attachment:integTest
==> Test Info: seed=9A749098650E4602; jvm=1; suite=1
==> Test Summary: 1 suite, 4 tests
:plugins:ingest-attachment:integTest#stop
:plugins:ingest-attachment:check

BUILD SUCCESSFUL

Total time: 2 mins 11.608 secs```

**Comment by dadoonet** [comment]
So, here is what I did (from elasticsearch root dir):

```sh
gradle clean
gradle install
```

It gives: `BUILD SUCCESSFUL`

Then:

```sh
cd plugins/ingest-attachment
gradle check
```

It now works. o_O

So I don't know what happened before that. The only difference between my first test and this one is that in the meantime I ran `gradle :plugins:ingest-attachment:check`.

Closing it. At least it will help it someone else hit this...

Thanks for your feedback @s1monw.
